### DISCUSSION TITLE
```required``` properties + Dependency Injection = Awesomeness

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#7037] [Show and tell]
Hi C# Team,
I just wanted to let you know that the work you did with ```required``` properties is even more awesome now that AutoFac supports it.

https://docs.autofac.org/en/latest/register/prop-method-injection.html#required-properties

It seriously saves a TON of boilerplate code since I no longer have to pass tons of parameters though the constructor hierarchy.

Thanks again for making this possible!

**Comment by 333fred** [comment]
Thanks for sharing Tony!
